src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java

```diff
@@ -0,0 +1,1202 @@
+                os.append(achIncomeNote.getNoteText());
+                os.append("\n");
+            }
+            os.close();
+
+            Note note = documentService.createNoteFromDocument(document, "");
+            String attachmentType = null;
+            BufferedInputStream fileStream = new BufferedInputStream(new FileInputStream(attachmentFile));
+            File file = new File(attachmentFile);
+            Attachment attachment = attachmentService.createAttachment(document.getDocumentHeader(), file.getName(), "text", (int) file.length(), fileStream, attachmentType);
+            note.setAttachment(attachment);
+            attachment.setNote(note);
+            noteService.save(note);
+        } catch (Exception e) {
+            LOG.error("Error while adding notes to advance deposit documents: " + e.getMessage(), e);
+            throw new RuntimeException("Error while adding notes to the document " + document.getDocumentNumber());
+        } finally {
+            if (os != null) {
+                try {
+                    os.close();
+                } catch (IOException ie) {
+                    // not really much to do at this point
+                }
+            }
+        }
+    }
+
+    protected void createSourceAccountingLine(AchIncomeTransaction transaction, AdvanceDepositDocument advanceDepositDocument) {
+        String chart = parameterService.getParameterValueAsString(GenerateAdvanceDepositDocumentsStep.class, CuFPParameterConstants.AdvanceDepositDocument.CHART);
+        String objectCode = parameterService.getParameterValueAsString(GenerateAdvanceDepositDocumentsStep.class, CuFPParameterConstants.AdvanceDepositDocument.OBJECT_CODE);
+        String account = parameterService.getParameterValueAsString(GenerateAdvanceDepositDocumentsStep.class, CuFPParameterConstants.AdvanceDepositDocument.ACCOUNT);
+
+        IncomingWireAchMapping matchingIncomingWireAchMapping = null;
+        Collection<IncomingWireAchMapping> incomingWireAchMappings = businessObjectService.findAll(IncomingWireAchMapping.class);
+
+        for (IncomingWireAchMapping mapping : incomingWireAchMappings) {
+            List<AchIncomeNote> notes = transaction.getNotes();
+            if (doNotesMatch(mapping, notes)) {
+                matchingIncomingWireAchMapping = mapping;
+                break;
+            }
+        }
+
+        if (ObjectUtils.isNotNull(matchingIncomingWireAchMapping)) {
+            chart = matchingIncomingWireAchMapping.getChartOfAccountsCode();
+            objectCode = matchingIncomingWireAchMapping.getFinancialObjectCode();
+            account = matchingIncomingWireAchMapping.getAccountNumber();
+        }
+
+        setupSourceAccountingLine(transaction, advanceDepositDocument, chart, objectCode, account);
+    }
+
+    protected boolean doNotesMatch(IncomingWireAchMapping mapping, List<AchIncomeNote> notes) {
+        final String regex = ".*" + mapping.getShortDescription() + ".*";
```

Comment:
If the ACH mapping description could potentially contain special regex characters and is not meant for having the characters treated as such, then it would be good to escape them before searching. Alternatively, since this code appears to just be checking if the description is present in the note, perhaps the regex matching could be replaced with a String indexOf() search.